services/provider/server/server.go

```diff
@@ -6,65 +6,214 @@ import (
 	"net"
 	"os"
 
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
 	"github.com/red-hat-storage/ocs-operator/services/provider/common"
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	TicketAnnotation = "ocs.openshift.io/provider-onboarding-ticket"
 )
 
 type ocsProviderServer struct {
 	pb.UnimplementedOCSProviderServer
+	client    client.Client
+	consumer  *ocsConsumer
+	namespace string
+}
+
+func NewOCSProviderServer(ctx context.Context, namespace string) (*ocsProviderServer, error) {
+	client, err := newClient()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new client. %v", err)
+	}
+
+	consumer, err := NewOCSConsumer(ctx, client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new OCSConumer instance. %v", err)
+	}
+
+	return &ocsProviderServer{
+		client:    client,
```

Comment:
This `client` is not used now. But will be needed when we call rook resources to generate the connection info.